### ISSUE TITLE
Platform aware widgets

### ISSUE THREAD

**Issue by Fayiz-A** [flutter/flutter#71140]
## Use case

The flutter team has done well for nativeness in Android and IOS. However some widgets like the alert dialog box are rendered only according to the design the developer is using such as Material or Cupertino in both the platforms. 

What I am saying is just by using a widget, the alert dialog box will be displayed like this

**IN ANDROID**
<img width="513" alt="Screenshot 2020-11-24 at 13 06 53" src="https://user-images.githubusercontent.com/68380206/100062657-f4420f80-2e55-11eb-9aff-5567e03043ef.png">

**IN IOS**
<img width="264" alt="Screenshot 2020-11-24 at 13 06 33" src="https://user-images.githubusercontent.com/68380206/100062672-fad08700-2e55-11eb-87d4-8560259142b6.png">

Instead of specifying the alert dialog box separately in Material App or Cupertino App. 
 
This will give a more native look.

In short, I want to say please introduce Platform aware widgets, which means just by writing AlertDialog, it will be rendered according to the Cupertino design in the IOS and according to the Material design in Android.

Thanks

**Comment by iapicca** [comment]
@Fayiz-A 
why don't you  use [dart:io ](https://api.dart.dev/stable/2.10.4/dart-io/dart-io-library.html) ?

```dart
import 'dart:io' show Platform;

Widget get myPlatformAdaptiveWidget {
  if (Platform.isIOS) {
    return MyCupertinoWidget();
  } else if (Platform.isAndroid) {
    return MyAndroidWidget();
  } else {
     return MyWebWidget();
  }
}
```

**Comment by pedromassangocode** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/33830.
Please see the design doc https://docs.google.com/document/d/14mxhVeVk1lClPkBi6VmB260_7fKUlZTM-Vn1bOres7M/edit

**Comment by Fayiz-A** [comment]
@pedromassangocode thanks
